app/server/lib/FlexServer.ts

```diff
@@ -600,6 +629,34 @@ export class FlexServer implements GristServer {
       if (isParameterOn(req.query.ready)) {
         checks.set('ready', this._isReady);
       }
+      if (isParameterOn(req.query.allInstancesReady)) {
+        const requestId = uuidv4();
+        const client = this._pubSubManager.getClient();
+
+        // If there is no redis, then our current instance is the only instance
+        const allInstances = await client?.smembers('grist-instances') || [this._instanceId];
```

Comment:
An early comment, not really a review: can all the new logic about live in a dedicated file? FlexServer is very big and does very many things already.

Reply:
I was thinking the same thing. Kind of want to move all of the health check logic somewhere else, not just the new part. I'll follow up with that.

Reply:
The new part seems very self-contained, and not overlapping with other health checks. So I still encourage a dedicated file for it.

Reply:
Okay, done.